docs/endpoint.md

```diff
@@ -5,14 +5,119 @@ The endpoint is one of the components provided by Tangle-accelerator, running on
+
 ```
-$ bazel build //endpoint:wp7702
-$ bazel build //endpoint:sim
+$ wget https://downloads.sierrawireless.com/tools/leaf/leaf_latest.deb -O /tmp/leaf_latest.deb && sudo apt install /tmp/leaf_latest.deb
 ```
 
-## HTTPS Connection Support
-The endpoint uses http connection as default. The message which sent to tangle-accelerator has been encrypted. So the HTTP connection would not be unsafe. To build with https connection support, add `--define https=enable` option.
+Create a workspace
+```
+mkdir -p workspace
+$ cd workspace
+```
+
+Setup the wp77xx target profile
+```
+leaf setup legato-latest -p swi-wp77_3.3.0
+```
+
+Finally, make the wp77xx endpoint target. Be careful the tangle-accelerator directory should be located at the sub level of the workspace directory created by `leaf setup legato-latest -p swi-wp77_3.3.0`
+```
+$ git clone https://github.com/DLTcollab/tangle-accelerator.git
+$ cd tangle-accelerator
+$ make TARGET=wp77xx legato // build endpoint as wp77xx target
+$ make ENDPOINT_TEST_ENABLE=true TARGET=wp77xx legato // build endpoint as wp77xx target in test mode
+```
+
+### How to build endpoint as x86 test target
+
+Install required packages:
+```
+$ sudo apt-get install -y   \
+    autoconf                \
+    automake                \
+    bash                    \
+    bc                      \
+    bison                   \
+    bsdiff                  \
+    build-essential         \
+    chrpath                 \
+    cmake                   \
+    cpio                    \
+    diffstat                \
+    flex                    \
+    gawk                    \
+    gcovr                   \
+    git                     \
+    gperf                   \
+    iputils-ping            \
+    libbz2-dev              \
+    libcurl4-gnutls-dev     \
+    libncurses5-dev         \
+    libncursesw5-dev        \
+    libsdl-dev              \
+    libssl-dev              \
+    libtool                 \
+    libxml2-utils           \
+    ninja-build             \
+    python                  \
+    python-git              \
+    python-jinja2           \
+    python-pkg-resources    \
+    python3                 \
+    texinfo                 \
+    unzip                   \
+    wget                    \
+    zlib1g-dev
+
+```
+
+Create a workspace
 ```
-$ bazel build --define https=enable //endpoint:wp7702
+mkdir -p workspace
+$ cd workspace
+```
+
+Download legato by git-repo
+```
+$ repo init -u git://github.com/legatoproject/manifest
+$ repo sync
+```
+
+You can also clone for specific version
+```
+$ repo init -u git://github.com/legatoproject/manifest -m legato/releases/19.07.0.xml
+$ repo sync
+```
+
+Build legato as localhost target
+```
+$ cd legato
+$ make localhost
+```
+
+Checkout the shell to legato shell
+```
+source ./framework/tools/scripts/configlegatoenv
+```
+
+Finally, go back to the TA project. Use GNU Make to make the endpoint project at the TA's root directory. 
```

Comment:
Finally, go back to the project root directory of tangle-accelerator. Use GNU Make to build the endpoint project under project root directory of tangle-accelerator.